data/topic-schema/src/test/kotlin/net/corda/schema/SchemaTests.kt

```diff
@@ -0,0 +1,63 @@
+package net.corda.schema
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.fasterxml.jackson.module.kotlin.KotlinFeature
+import com.fasterxml.jackson.module.kotlin.KotlinModule
+import com.fasterxml.jackson.module.kotlin.readValue
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import java.io.File
+import kotlin.reflect.full.companionObject
+
+class SchemaTests {
+    // Setup jackson mapper to support yaml with null values
+    private val mapper = ObjectMapper(YAMLFactory()).registerModule(
+        KotlinModule.Builder()
+            .withReflectionCacheSize(512)
+            .configure(KotlinFeature.NullToEmptyCollection, true)
+            .configure(KotlinFeature.NullToEmptyMap, true)
+            .configure(KotlinFeature.NullIsSameAsDefault, false)
+            .configure(KotlinFeature.SingletonSupport, false)
+            .configure(KotlinFeature.StrictNullChecks, false)
+            .build()
+    )
+
+    @Suppress("UNCHECKED_CAST")
+    private val yamlFileData by lazy {
+        // Scan resources in classpath to find all the yaml files to scan
+        this::class.java.classLoader.getResources("net/corda/schema")
+            .toList()
+            .filterNotNull()
+            .map { File(it.file) }
+            .filter { it.isDirectory }
+            .flatMap { it.listFiles()!!.toList() }
+            .filter { it.name.endsWith("yaml") || it.endsWith("yml") }
+            .map { mapper.readValue(it.readText()) as Map<String, *> }
+            as List<Map<String, Map<String, Map<String, *>>>>
+    }
+
+    // Scan companion objects for constant definitions
+    private val memberMap by lazy {
+        val schemaCompanions = Schemas::class.nestedClasses.mapNotNull {
+            it.companionObject
+        }
+        val members = schemaCompanions.flatMap {
+            it.members
+        }.filter { it.isFinal }
+        val memberValueMap = members.map {
+            it.name to it.call(emptyList<Any>()) as String
+        }
+
+        memberValueMap.toMap()
+    }
+
+    @Test
+    fun `Validate that all yaml topics have matching code value`() {

```

Comment:
Longer term we would want this to heck that each section has right subset of the topics

Reply:
NP, I should be able to parameterize the test